### ISSUE TITLE
Installation issues with new ASP.Net application and version 1.6

### ISSUE THREAD

**Issue by PureDevelopers** [leen3o/mvcforum#10]
Step 1: Open Visual Studio 2013
Step 2: Creat a new asp.net project of type MVC with unit tests and individual user accounts.
Step 3: Use package manager to update all packages using stable only.
Step 4: Used the package manager console to install the forum.
Step 5: Run app in debug mode with the following result:
I was instructed to install the forum db, which I did, but then got an error message about not being able to copy the database file. I went through a bunch of hoops to make sure the db was not created, but couldn't resolve the issue, so I decided that maybe I need to run the debugger so that the db is created before I install the forum, so I:
repeated steps 1 through 3
Then I ran the app in debug and registered which created the db for me.
Then I stopped the debugger and performed steps 4 and 5.
Now I am getting the following error: 
"Multiple types were found that match the controller named 'Home'. This can happen if the route that services this request ('{controller}/{action}/{id}') does not specify namespaces to search for a controller that matches the request. If this is the case, register this route by calling an overload of the 'MapRoute' method that takes a 'namespaces' parameter.
The request for 'Home' has found the following matching controllers:
MVCForum.Website.Controllers.HomeController
BeachVolleyballWorldForum.Controllers.HomeController"

Do you provide installation instructions for creating a new asp.Net application, and then installing the forum?
Does the forum work correctly with the latest Nuget packages?
When I install the forum software, the web.config file is overwritten, which replaces the DefaultConnection string with a forum connection string, but the ApplicationDbContext for Identity is still pointed at DefaultConnection???

**Comment by PureDevelopers** [comment]
Also, is this forum software integrated with the new Asp.Net identity 2.2? does it require a separate login? What about third party providers?

**Comment by ryios** [comment]
I don't know about Asp.Net identity 2.2.  However if my pull request goes through I implemented WIF (Windows Identity Foundation) from .Net 4.5 into MVC Forum yesterday and that handles external login sources.  But you need to build a SecurityTokenService Issuer Login Site for it to authenticate remotely against.  The source code I'm pull requesting on also contains an example STS Issuer Site.

WIF is just something I know a lot about, and also how SharePoint does all of it's authentication and I have 4 years SP experience, so was easiest for me to use for SSO.  In my opinion WIF is a strong capable auth framework as well and is Claims Based, so if Microsoft doesn't make it available in .Net Core 5 I'll be a little disappointed.

Asp.Net Identity 2.2 seems to be what they plan on having in ASP.Net VNext on .Net Core 5 though, so I'll pick that up and learn it when I get time and maybe make additional integrations.

--------
As to your installation error, The specific template you stated already has Views for Account/Home/Manage/Shared etc, and will conflict with MVCForum.

MVCForum is designed, more or less, to be it's own Project in Visual Studio.  Installing it in an existing Project isn't going to work.  Also that template appears to be using Identity Manager, and looking at it, I'm pretty sure I could make it work.. But here's the thing,

MVCForum has it's own users table in it's database, which has foreign keys all over the place.  E.g. it needs to have it's own user table.  MVCForum supports OAuth through FB and Gmail, but the way it works is when a user logs in with Facebook, MVCForum creates them a user account in MVCForum on the fly with no password, same thing for Gmail.  The WIF integration I just committed also does the same thing.  So if I implemented Identity 2.2 it would work the same way as well.

Short Version- Users will never not be stored in MVC, the only real question is if they are used for authentication or not.  Via Forms authentication (default MVC logon mode) they are.  Via WIF authentication comes from the STS.  Via Facebook they come from Facebook, etc for Gmail.
